src/main/java/io/xdb/core/communication/Communication.java

```diff
@@ -50,4 +55,13 @@ public void publishToLocalQueue(final String queueName, final String dedupId, fi
 
         localQueueMessagesToPublish.add(message);
     }
+
+    /**
+     * Trigger new state decision from @RPC methods.
+     *
+     * @param stateDecision     the state decision to trigger.
+     */
+    public void triggerNewStateDecision(final StateDecision stateDecision) {
```

Comment:
Can you propose any better names? If no, let's keep using the current one.

Reply:
I meant you can just provide only:
```
communication.triggerStateExecution(...)
```
for now and NOT allowing complete/fail process. It's not really critical to provide that because:
* User can call "StopProcess" to terminate
* User can send a local Queue message to do it on the background
* It's not yet a important use case to provide it. If it becomes necessary, we can provide like `communication.gracefulComplete(....)` in the future
* You can keep the same implementation in server

Reply:
The server alraedy supports closing thread in RPC, and I think it no harm to enable it in SDK too. 

As for the naming, maybe just `triggerStateDecision` by removing the word `new`.

Reply:
The reasons why I am not a big fan of that:
* there is now state here to make a decision. An RPC doesn’t below to any state. The decision here is for the whole process to either stop the process, or start new states 
* it’s not an important use case for setting this abitrary decision 
* The most important use case is to let user trigger new state execution in the background, because rpc is foreground , having able to trigger a background execution is very useful. 
* therefore, triggerStateExecutuon( stateClass, input ) will be much more friendly to use

Reply:
> there is now state here to make a decision
I think a decision does not necessarily have to be made in a state. In RPC, we can also use state decision to start states or stop process. This is what has been implemented currently.

> The most important use case
I am not sure. We at least need some user data to analyze and draw the conclusion. If your point is more on the API level, then I think it's still conveient to trigger state execution by `triggerStateDecision(StateDecision.singleNestState(xxx))`

Reply:
> We at least need some user data to analyze and draw the conclusion.

ahh that's what I wanted to say too. 

Since we are not sure, the safest/best way is to expose minimum APIs for now. As a general rule for SDK API design, the more we exposed, the harder we can change in the future. 
It's never late to add this `triggerStateDecision` or `triggerRPCDecision` in the future if a user requesting for it with a concrete reason.

Reply:
I think we can expose as many completed funtions as possible so that users can choose which functionality they like.

As for the SDK API naming, maybe you can read the current API for AsyncState execute first, then look back to this one? I don't wanna make the two inconsistent for users.

Reply:
Why we should expose as many complex features as possible? 

I did read the APIs because I created them in IWF. I understand that rpc is not the same as state execute. But they are different by nature. 

In fact, this communication object should have two different ones, one for state and one for rpc.